### PR TITLE
Implemented transactionAt

### PR DESCRIPTION
**Pull request by jonathanBuhler**

### CODE REVIEW

**Comment by raulk** [comment]
Hey Jonathan, could you add some tests for this functionality? 
1) an index that exists in the requested block
2) an index that doesn’t exist in the requested block 
3) a negative index

File: src/__tests__/core/block.test.ts [hunk 1]
```diff
@@ -15,6 +15,56 @@ test('block: select single block by number', async () => {
...
 });
 
+test('block: select single transaction from block by index', async () => {
+  const query =  `
+    {
+        block(number: 5771878) {
+          difficulty
+          transactionAt(index: 0) {
+            inputData
+            status
```
Comment (akhila-raju) [inline src/__tests__/core/block.test.ts]: For lines 24 and 25, check a different field for value rather than a null field. Hash is a good one to check.

Comment (akhila-raju) [reply src/__tests__/core/block.test.ts]: Same goes for the tests below

File: src/__tests__/core/block.test.ts [hunk 2]
```diff
@@ -15,6 +15,56 @@ test('block: select single block by number', async () => {
...
+          difficulty
+          transactionAt(index: -1) {
+            inputData
+            status
+        }
+      }
+    }
+  `;
+
+  const expected = { data: { block: { difficulty: 3351918027816898, transactionAt: null } } }
```
Comment (akhila-raju) [inline src/__tests__/core/block.test.ts]: For this test and the one below, an error should be thrown indicating why it's returning null

File: src/resolvers/block.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ const block: IFieldResolver<any, any> = async (obj, args: IBlockArgs) => {
     throw new Error('Only one of number, hash or tag argument should be provided.');
   }
 
-  const block = await web3.eth.getBlock(params[0]);
+  const block = await web3.eth.getBlock(params[0], true);
```
Comment (akhila-raju) [inline src/resolvers/block.ts]: What was the reason for this change?

Comment (jonathanBuhler) [reply src/resolvers/block.ts]: When false (default) getBlock only returns a block with transaction hashes but when true it returns a block with the transaction objects, which are needed by transactionAt.

Comment (akhila-raju) [reply src/resolvers/block.ts]: LGTM - Raul will review and approve :)

File: src/__tests__/core/block.test.ts [hunk 4]
```diff
@@ -15,6 +15,57 @@ test('block: select single block by number', async () => {
   expect(result).toEqual(expected);
 });
 
+test('block: select single transaction from block by index', async () => {
+  const query = `
+  {
+    block(number: 5771878) {
```
Comment (raulk) [inline src/__tests__/core/block.test.ts]: @jonathanBuhler @akhila-raju (Just a nit) Generally I would suggest using lower block numbers in tests, the reason being that chain reorganisations are possible at any time, and the lower the block number the higher the guarantee that our test data will remain invariant.

File: src/resolvers/block.ts (hunk 3 above)
Comment (raulk) [reply src/resolvers/block.ts]: Looks good, yeah.

**Comment by raulk** [comment]
[ch99] -- @jonathanBuhler this is the Clubhouse task number. Adding this as a comment will make Clubhouse move the ticket forward automatically.

**Pull request merged**